application/comit_node/tests/htlc_harness/erc20_harness.rs

```diff
@@ -32,7 +32,19 @@ impl Default for Erc20HarnessParams {
         Self {
             alice_initial_ether: EtherQuantity::from_eth(1.0),
             htlc_timeout: HTLC_TIMEOUT,
-            htlc_secret: SECRET.clone(),
+            htlc_secret_hash: Secret::from_vec(SECRET).unwrap().hash(),
+            alice_initial_tokens: U256::from(1000),
+            htlc_token_value: U256::from(400),
+        }
+    }
+}
+
+impl From<SecretHash> for Erc20HarnessParams {
+    fn from(secret_hash: SecretHash) -> Self {
```

Comment:
❌ Are you sure you want to misuse the `from` trait ? I would rather like a normal function for that

Reply:
How is it a misuse? I am creating a Erc20HarnessParams from a `SecretHash` has it is the only dynamic parameter.

Reply:
`From` should convert from one type into another, i.e. the counterpart of `into`.  It gives the impression that all necessary information is given in `secret_hash` which is not the case. The present functionality is more a mix of `default` with a specific `secret_hash` .

Reply:
Yes, all necessary information **is** given in `secret_hash`. It is the case.

Reply:
Ok, I yield.